crates/fleet-base/src/fleetdata.rs

```diff
@@ -73,16 +72,14 @@ pub struct FleetSharedSecret {
 }
 
 /// Returns None if recipients.is_empty()
-pub fn encrypt_secret_data(
-	recipients: impl IntoIterator<Item = impl Recipient + Send + 'static>,
+pub fn encrypt_secret_data<'a>(
+	recipients: impl IntoIterator<Item = &'a (impl Recipient + Send + 'static)>,
```

Comment:
Looks fishy, can you simplify this associated type?

Reply:
Whatever with_recipients accepts should be fine I think.

Reply:
Simplified it to `Item = &'a dyn Recipient` and changed the callers to reflect the change